app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java

```diff
@@ -28,22 +34,79 @@ protected void onCreate(Bundle savedInstanceState) {
+      mEditTextMessage = (EditText) findViewById(R.id.txt_message);
 
-        mButtonSend.setOnClickListener(new View.OnClickListener() {
+      mButtonSend.setOnClickListener(new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+          //We get the number and the message that were inserted in the view
+          String phoneNro = mEditTextTelephone.getText().toString();
+          String smsText = mEditTextMessage.getText().toString();
+
+          String sent = "SMS_SENT";
+          String delivered = "SMS_DELIVERED";
+
+          PendingIntent sentPI = PendingIntent.getBroadcast(getApplicationContext(), 0,
+            new Intent(sent), 0);
+
+          PendingIntent deliveredPI = PendingIntent.getBroadcast(getApplicationContext(), 0,
+            new Intent(delivered), 0);
+
+          //---when the SMS has been sent---
+          registerReceiver(new BroadcastReceiver(){
             @Override
-            public void onClick(View v) {
-                Intent smsIntent = new Intent(getBaseContext(), SmsIntentService.class);
-                Bundle smsData = new Bundle();
-                smsData.putString(SmsIntentService.SMS_MESSAGE, mEditTextMessage.getText().toString());
-                smsData.putString(SmsIntentService.SMS_PHONE, mEditTextTelephone.getText().toString());
-                smsIntent.putExtras(smsData);
-
-                startService(smsIntent);
+            public void onReceive(Context arg0, Intent arg1) {
+              switch (getResultCode())
+              {
+                case Activity.RESULT_OK:
+                  Toast.makeText(getBaseContext(), "SMS sent",
+                    Toast.LENGTH_SHORT).show();
+                  break;
+                case SmsManager.RESULT_ERROR_GENERIC_FAILURE:
+                  Toast.makeText(getBaseContext(), "Generic failure",
+                    Toast.LENGTH_SHORT).show();
+                  break;
+                case SmsManager.RESULT_ERROR_NO_SERVICE:
+                  Toast.makeText(getBaseContext(), "No service",
+                    Toast.LENGTH_SHORT).show();
+                  break;
+                case SmsManager.RESULT_ERROR_NULL_PDU:
+                  Toast.makeText(getBaseContext(), "Null PDU",
+                    Toast.LENGTH_SHORT).show();
+                  break;
+                case SmsManager.RESULT_ERROR_RADIO_OFF:
+                  Toast.makeText(getBaseContext(), "Radio off",
+                    Toast.LENGTH_SHORT).show();
+                  break;
+              }
             }
-        });
+          }, new IntentFilter(sent));
+
+          //---when the SMS has been delivered---
+          registerReceiver(new BroadcastReceiver(){
+            @Override
+            public void onReceive(Context arg0, Intent arg1) {
+              switch (getResultCode())
+              {
+                case Activity.RESULT_OK:
+                  Toast.makeText(getBaseContext(), "SMS delivered",
```

Comment:
Same here, I think would be good to place "SMS delivered" inside a constants file.